src/library/basetracktablemodel.cpp

```diff
@@ -362,6 +363,30 @@ int BaseTrackTableModel::columnCount(const QModelIndex& parent) const {
     return countValidColumnHeaders();
 }
 
+void BaseTrackTableModel::cutTracks(const QModelIndexList& indices) {
+    copyTracks(indices);
+    removeTracks(indices);
```

Comment:
Just a thought: 
My expectation was that the removeTracks is delayed until the past was successful. 

On the other hand this is no file manager and when doing it in a text editor the text is immediately removed. 
But there is a undo feature Ctrl-Z ..

Did you consider this?

Reply:
I agree this doesn't match the usual file manager behaviour.
However, tracks are removed on Cut, not just greyed out as in file managers, hence I think the current solution provides predictable UX.

FWIW cut & paste works reliably as Undo, i.e. tracks positions in playlists are properly restored regardless the sorting.

Reply:
Implementing a proper undo should be feasible though and pretty simple if it's just one step deep.